### PR TITLE
Add e2e for darwin and linux on gh-action

### PR DESCRIPTION
**Pull request by wzshiming**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        shell: bash
+        run: |
+          ./hack/e2e-test.sh kwokctl/kwokctl_${{ matrix.kwokctl-runtime }}_snapshot
+
+      - name: Test Benchmark
+        shell: bash
+        if: matrix.kwokctl-runtime != 'kind'
+        continue-on-error: true
+        run: |
+          ./hack/e2e-test.sh kwokctl/kwokctl_${{ matrix.kwokctl-runtime }}_benchmark
```
Comment (Huang-Wei) [inline .github/workflows/test.yaml:108]: Is that we're going to use GH action test kwokctl, and use k8s test-infra to test kwok?

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        shell: bash
+        run: |
+          ./hack/requirements.sh kind
+          kind version
+
+      - name: Docker Info
+        if: ${{ matrix.kwokctl-runtime != 'binary' }}
+        shell: bash
+        run: |
+          docker info
```
Comment (yibozhuang) [inline .github/workflows/test.yaml:85]: what's the purpose of this? just for logging the docker version?

File: .github/workflows/test.yaml [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+      - name: Test Snapshot
+        shell: bash
+        run: |
+          ./hack/e2e-test.sh kwokctl/kwokctl_${{ matrix.kwokctl-runtime }}_snapshot
+
+      - name: Test Benchmark
+        shell: bash
+        if: matrix.kwokctl-runtime != 'kind'
+        continue-on-error: true
```
Comment (yibozhuang) [inline .github/workflows/test.yaml:106]: why continue on error here and not the other tests?

File: .github/workflows/test.yaml (hunk 1 above)
Comment (wzshiming) [reply .github/workflows/test.yaml:108]: Ooops I seem to have missed Kwok's e2e

File: .github/workflows/test.yaml (hunk 3 above)
Comment (wzshiming) [reply .github/workflows/test.yaml:106]: This often times out in k8s test-infra, depending on machine performance

File: .github/workflows/test.yaml (hunk 2 above)
Comment (wzshiming) [reply .github/workflows/test.yaml:85]: Yes

**Review by liangyuanpeng — APPROVED** [review]

**Pull request merged**